### PR TITLE
Fixes slipping on tables

### LINKED ISSUE #5965: Slipping on foam when climbing onto a table with foam underneath
If the janitor lets off a cleaning foam grenade, and you're adjacent to a table with foam underneath it, climbing onto the table causes you to slip, as though walking on the foam underneath it.

### LINKED ISSUE #12275: Walking on a table with foam under it slips you.
I don't know whether this works with banana peels or the like, if you somehow get them under a table.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #5965
Fixes #12275

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -228,6 +228,9 @@
 	if(has_gravity(src))
 		var/obj/buckled_obj
 		var/oldlying = C.lying
+		var/obj/structure/table/T = locate(/obj/structure/table,src)
```
Comment (KazeEspada) [inline code/game/turfs/turf.dm]: SNOOOOOOOOOOOOWFLAKE

**Comment by swankcookie** [comment]
This is a sprite layer issue, IMO. Tables that are hit by foam grenades should be slippery.

**Comment by Aranclanos** [comment]
doesn't meet the standards :-1:

**Comment by AnturK** [comment]
What standard ? Unless you're talking about the second locate path.

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (AnturK) [inline code/game/turfs/turf.dm]: Checking for all dense objects is bound to have some issues

**Comment by Aranclanos** [comment]
checking if there's a table each time someone slips is no acceptable

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (duncathan) [inline code/game/turfs/turf.dm]: This is so bad my god

**Comment by AnturK** [comment]
Alternative being what exactly ?

**Comment by phil235** [comment]
The check is pretty snowflakey, yeah.
We can always go back to putting foam layer above table layer by reverting #11547 (which reverted my change in the first place).

**Comment by AnturK** [comment]
Alright, my alternative idea wasn't as good. Would mean adding new var on the base turf which is even worse imo

**Comment by swankcookie** [comment]
Once again, the slipping is not the issue. the fact that foam doesn't cover the table is, and would be easily fixed with a layer attribute.

**Comment by AnturK** [comment]
This isn't much better but at least it's extendable i guess ?

**Comment by phil235** [comment]
You need to also take into account small mobs who move under the table (mouse, hidding larva)

**Comment by Aranclanos** [comment]
still don't like it

**Comment by Dorsisdwarf** [comment]
I thought foam WAS above tables but it was removed for being uglly as sin?

**Comment by Razharas** [comment]
Just make foam be above things again
Argument that foam hidesthings is retarded since vapor clouds block vision completely, using these things is supposed to be obstructive

**Comment by phil235** [comment]
If people want  to see through the foam we could just make it slightly transparent.

**Comment by Razharas** [comment]
Or this yes

**Comment by AnturK** [comment]
I dunno, i think it really looks better under the tables. But i can't think of anything faster than linear scan of contents.

**Comment by xxalpha** [comment]
>If people want to see through the foam we could just make it slightly transparent.

Genius.

**Comment by Razharas** [comment]
Fixed by xxalphas linked pr

**Pull request closed without merging**